### PR TITLE
 Allow random parameters to be generated from correlated distribution (fixes #533) 

### LINKED ISSUE #533: Allow random parameters to be generated from correlated distribution
The idea would be to allow correlation between two or more random parameters

```r
my_randoms <- list(
  act.rate = param_random(c(0.25, 0.5, 0.75)),
  tx.prob = function() rbeta(1, 1, 2),
  stratified.test.rate = function() c(
    rnorm(1, 0.05, 0.01),
    rnorm(1, 0.15, 0.03),
    rnorm(1, 0.25, 0.05)
  ),
  list(
    params = c("p1", "p2", "p3"),
    generator = function() list(runif(1), rbinom(1, 10, 0.3), rnorm(1))
  )
)
```

### LINKED ISSUE #553: Duplicated fixed and random parameter printing with print.param.net before netsim
In this example:

```r
## Example SIR model parameterization with fixed and random parameters
# Network model estimation
nw <- network_initialize(n = 100)
formation <- ~edges
target.stats <- 50
coef.diss <- dissolution_coefs(dissolution = ~offset(edges), duration = 20)
est <- netest(nw, formation, target.stats, coef.diss, verbose = FALSE)

# Random epidemic parameter list (here act.rate values are sampled uniformly
# with helper function param_random, and inf.prob follows a general Beta
# distribution with the parameters shown below)
my_randoms <- list(
  act.rate = param_random(1:3),
  inf.prob = function() rbeta(1, 1, 2)
)

# Parameters, initial conditions, and control settings
param <- param.net(rec.rate = 0.02, random.params = my_randoms)

# Printing parameters shows both fixed and and random parameter functions
param
```

`act.rate` is printed as both a fixed and a random parameter. It should only print as a random parameter.

The problem does not happen post-netsim:

```r
# Set initial conditions and controls
init <- init.net(i.num = 10, r.num = 0)
control <- control.net(type = "SIR", nsteps = 10, nsims = 3, verbose = FALSE)

# Simulate the model
sim <- netsim(est, param, init, control)

# Printing the sim object shows the randomly drawn values for each simulation
sim
```

Looks good.

### PR DESCRIPTION
**Pull request by AdrienLeGuillou**
Initial implementation to close #533 using the `param_random_set` approach.

the following code works:

```r
pkgload::load_all("../EpiModel")

## Example SIR model parameterization with fixed and random parameters
# Network model estimation
nw <- network_initialize(n = 100)
formation <- ~edges
target.stats <- 50
coef.diss <- dissolution_coefs(dissolution = ~offset(edges), duration = 20)
est <- netest(nw, formation, target.stats, coef.diss, verbose = FALSE)

# Defining a parameter generator function
generate_correlated_params <- function() {
  param.unique <- runif(1)
  param.set.1 <- param.unique + runif(2)
  param.set.2 <- param.unique * rnorm(3)

  return(list(param.unique, param.set.1, param.set.2))
}

correlated_params <- t(replicate(10, unlist(generate_correlated_params())))
correlated_params <- as.data.frame(correlated_params)
colnames(correlated_params) <- c(
  "param.unique",
  "param.set.1_1","param.set.1_2",
  "param.set.2_1", "param.set.2_2", "param.set.2_3"
)


param_random_set <- correlated_params
param <- list()


# Defining the list of random parameters
my_randoms <- list(
  act.rate = param_random(c(0.25, 0.5, 0.75)),
  tx.prob = function() rbeta(1, 1, 2),
  stratified.test.rate = function() c(
    rnorm(1, 0.05, 0.01),
    rnorm(1, 0.15, 0.03),
    rnorm(1, 0.25, 0.05)
  ),
  # for correlated parameters, we must define the name of the parameters and
  # the function that will produce them
  param_random_set = correlated_params
)

param <- param.net(rec.rate = 0.02, random.params = my_randoms)

p2 <- generate_random_params(param)

```

# To Do:

- [x] test for obvious edge cases
- [x] document this new implementation
- [x] modify the `print.param` function to correctly display these new parameters (must be finished in order to close #553 with #561)

### CODE REVIEW

File: vignettes/Random-parameterization-interface.Rmd [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+"Fixed Parameters" section: `inf.prob` and `groups`. The other ones are displayed
+under the "Random Parameters". `act.rate` and `dummy.param` now have 3 values associated
+with them, one per simulation. `dummy.strat.param` have `<list>` as value because
+each simulation has a vector of size 2.
+
+We can inspect the values with:
+
+``` {r generators_inspect}
+str(mod$param$random.params.values)
+```
```
Comment (smjenness) [inline vignettes/Random-parameterization-interface.Rmd:129]: note to update this when you have the extraction function ready

**Review by smjenness — APPROVED** [review]

**Comment by smjenness** [comment]
This looks great! Thanks @AdrienLeGuillou
